packages/trpc/server/routers/viewer/organizations/createTeams.handler.ts

```diff
@@ -238,6 +240,18 @@ async function moveTeam({
     teamSlug: newSlug,
     orgSlug: org.slug || (orgMetadata?.requestedSlug ?? null),
   });
+
+  // Cancel existing stripe subscriptions once the team is migrated
+  const parsedMetadata = teamMetadataSchema.safeParse(team.metadata);
+  if (parsedMetadata.success) {
+    const subscriptionId = parsedMetadata.data?.subscriptionId;
+    if (subscriptionId) {
+      log.debug("Canceling stripe subscription", safeStringify({ team, subscriptionId }));
+      await stripe.subscriptions.cancel(subscriptionId);
+    } else {
+      log.warn("No subscriptionId found in team metadata", safeStringify({ team, parsedMetadata }));
+    }
+  }
```

Comment:
I think we should atleast log the error in else case because it would be not normal to have a metadata that doesn't parse.

Also, how about introducing a function `getSubscriptionId` that abstracts out the metadata thing.
Code becomes more readable as 

```
const subscriptionId = getSubscriptionForTeam(team);
if (subscriptionId) {
      log.debug("Canceling stripe subscription", safeStringify({ team, subscriptionId }));
      await stripe.subscriptions.cancel(subscriptionId);
}

```